src/main/java/seedu/address/logic/parser/AddressBookParser.java

```diff
@@ -62,15 +69,56 @@ public Command parseCommand(String userInput) throws ParseException {
         case ListCommand.COMMAND_WORD:
             return new ListCommand();
 
-        case ExitCommand.COMMAND_WORD:
-            return new ExitCommand();
-
         case HelpCommand.COMMAND_WORD:
             return new HelpCommand();
 
+        case ExitCommand.COMMAND_WORD:
+            return new ExitCommand();
+
         default:
             throw new ParseException(MESSAGE_UNKNOWN_COMMAND);
         }
     }
 
+    /**
+     * Parses user input into commandWithFile for execution.
+     *
+     * @param userInput full user input string
+     * @return the commandWithFile based on the user input
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public static Optional<CommandWithFile> parseCommandWithFile(String userInput) throws ParseException {
+        final Matcher matcher = COMMAND_WITH_FILE_FORMAT.matcher(userInput.trim());
+        if (!matcher.matches()) {
+            return Optional.empty();
+        }
+        final String commandWord = matcher.group("commandWord");
+        final String objectType = matcher.group("objectType");
+        switch (commandWord) {
+        case ExportCommand.COMMAND_WORD:
+            switch (objectType) {
+            case ExportCommand.PROPERTIES:
+                return Optional.of(new ExportPropertiesCommand());
+            case ExportCommand.BUYERS:
+                return Optional.of(new ExportBuyersCommand());
```

Comment:
Is this consistent with how parsing is done by the regular parseCommand? The `objectType` in the regular parseCommand is retrieved from the preamble in ParserUtil.

Reply:
Created an ExportComnandParser